### ISSUE TITLE
fexpr~ fails to excute properly: count_1003 = $y2[-1] + 1 \; if(count_1003 > n_1003 \, 1 \, count_1003)

### ISSUE THREAD

**Issue by x37v** [x37v/jit-expr#1]
convert this to a pd file:
[test.txt](https://github.com/x37v/jit-expr/files/1716987/test.txt)

generated llvm assembly:

```
jit/fexpr~: 
 ; ModuleID = 'jit/expr' 
 source_filename = "jit/expr" 
 target datalayout = "e-m:e-i64:64-f80:128-n8:16:32:64-S128" 
  
 %union.input_arg_t = type \{ float* \} 
 %t_symbol_ptr = type opaque 
  
 define internal void @jitexpr(float** %out, %union.input_arg_t* %in, i32 %veclen) \{ 
 entry: 
   %0 = alloca float**, align 8 
   store float** %out, float*** %0, align 8 
   br label %loop 
  
 loop:                                             ; preds = %ifcont, %entry 
   %loopvar = phi i32 [ 0, %entry ], [ %nextvar, %ifcont ] 
   %1 = load float*, float** %out, align 8 
   %2 = sext i32 %loopvar to i64 
   %3 = getelementptr inbounds float, float* %1, i64 %2 
   %4 = getelementptr inbounds float*, float** %out, i64 1 
   %inputy1 = load float*, float** %4, align 8 
   %cast = sitofp i32 %veclen to float 
   %bottom = fsub float 0.000000e+00, %cast 
   %lttmp = fcmp ogt float %bottom, -1.000000e+00 
   %5 = select i1 %lttmp, float %bottom, float -1.000000e+00 
   %cast1 = sitofp i32 %loopvar to float 
   %offset = fadd float %5, %cast1 
   %ltmp = fcmp olt float %offset, 0.000000e+00 
   %6 = select i1 %ltmp, float %cast, float 0.000000e+00 
   %7 = fadd float %offset, %6 
   %cast3 = fptosi float %cast to i32 
   %tmparrayinterp = call float @jit_expr_array_read(float* %inputy1, float %7, i32 %cast3) 
   %tmpvalueassign = call float @jit_expr_value_assign(i64 94627192760480, float %tmparrayinterp) 
   %addtmp = fadd float %tmpvalueassign, 1.000000e+00 
   store float %addtmp, float* %3, align 4 
   %8 = load float*, float** %4, align 8 
   %9 = getelementptr inbounds float, float* %8, i64 %2 
   %tmpvalueget = call float @jit_expr_value_get(i64 94627192760480) 
   %tmpvalueget4 = call float @jit_expr_value_get(i64 94627193092816) 
   %gttmp = fcmp ogt float %tmpvalueget, %tmpvalueget4 
   br i1 %gttmp, label %ifcont, label %else 
  
 else:                                             ; preds = %loop 
   %tmpvalueget6 = call float @jit_expr_value_get(i64 94627192760480) 
   br label %ifcont 
  
 ifcont:                                           ; preds = %loop, %else 
   %iftmp = phi float [ %tmpvalueget6, %else ], [ 1.000000e+00, %loop ] 
   store float %iftmp, float* %9, align 4 
   %nextvar = add i32 %loopvar, 1 
   %10 = icmp eq i32 %nextvar, %veclen 
   br i1 %10, label %afterloop, label %loop 
  
 afterloop:                                        ; preds = %ifcont 
   ret void 
 \} 
  
 declare float @jit_expr_array_read(float*, float, i32) 
  
 declare float @jit_expr_value_assign(%t_symbol_ptr*, float) 
  
 declare float @jit_expr_value_get(%t_symbol_ptr*) 
```

**Comment by x37v** [comment]
I wonder if the `=` in `count_1003 = $y2[-1] + 1` takes precedence over the `+`
Or, maybe the if statement is being messed up by the loop

**Comment by x37v** [comment]
assignment was incorrect, that is fixed, but [jit/fexpr~ 1; 2; 3] outputs on the wrong outlet..

**Comment by x37v** [comment]
fixed in fa7bfc686fc1a7e09ae12e9960b0c73c45ce33f2
